examples/plot_vgm_error.py

```diff
@@ -0,0 +1,256 @@
+an area of interest (e.g, to estimate the error of a mean or sum of samples). Once the spatial correlations are quantified,
+several methods exist to derive the related measurement error integrated in space (`Rolstad et al. (2009) <https://doi.org/10.3189/002214309789470950>`_
+, Hugonnet et al. (in prep)). More details are available in :ref:`spatialstats`.
+
+Here, we show an example in which we estimate spatially integrated elevation measurement errors for a DEM difference of
+Longyearbyen glacier, demonstrated in :ref:`sphx_glr_auto_examples_plot_nuth_kaab.py`. We first quantify the spatial
+correlations using :func:`xdem.spatialstats.sample_multirange_variogram` based on routines of `scikit-gstat
+<https://mmaelicke.github.io/scikit-gstat/index.html>`_. We then model the empirical variogram using a sum of variogram
+models using :func:`xdem.spatialstats.fit_model_sum_vgm`.
+Finally, we integrate the variogram models for varying surface areas to estimate the spatially integrated elevation
+measurement errors using :func:`xdem.spatialstats.neff_circ`, and empirically validate the improved robustness of
+our results using `xdem.spatialstats.patches_method`, an intensive Monte-Carlo sampling approach.
+
+"""
+# sphinx_gallery_thumbnail_number = 6
+import matplotlib.pyplot as plt
+import numpy as np
+import xdem
+import geoutils as gu
+
+# %%
+# We start by loading example files including a difference of DEMs at Longyearbyen glacier and the outlines to rasterize
+# a glacier mask.
+# Prior to differencing, the DEMs were aligned using :class:`xdem.coreg.NuthKaab` as shown in
+# the :ref:`sphx_glr_auto_examples_plot_nuth_kaab.py` example. We later refer to those elevation differences as *dh*.
+
+dh = xdem.DEM(xdem.examples.get_path("longyearbyen_ddem"))
+glacier_outlines = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+mask_glacier = glacier_outlines.create_mask(dh)
+
+# %%
+# We remove values on glacier terrain in order to isolate stable terrain, our proxy for elevation measurement errors.
+dh.data[mask_glacier] = np.nan
+
+# %%
+# We estimate the average per-pixel elevation measurement error on stable terrain, using both the standard deviation
+# and normalized median absolute deviation. For this example, we do not account for the non-stationarity in elevation
+# measurement errors quantified in :ref:`sphx_glr_auto_examples_plot_nonstationary_error.py`.
+print('STD: {:.2f}'.format(np.nanstd(dh.data))+' meters.')
+print('NMAD: {:.2f}'.format(xdem.spatialstats.nmad(dh.data))+' meters.')
+
+# %%
+# The two measures of dispersion are quite similar showing that, on average, there is a small influence of outliers on the
+# elevation differences. The per-pixel precision is about :math:`\pm` 2.5 meters.
+# **Does this mean that every pixel has an independent measurement error of** :math:`\pm` **2.5 meters?**
+# Let's plot the elevation differences to visually check the quality of the data.
+plt.figure(figsize=(8, 5))
+plt_extent = [
```

Comment:
You can do it all in one-line with:
```
plt.figure(figsize=(8, 5))
dh.show(ax=plt.gca(), cmap='RdYlBu', vmin=-4, vmax=4, cb_title='Elevation differences (m)')
```
(ok two lines are needed if you want to set the figure size...)